core/src/main/java/org/jruby/RubyModule.java

```diff
@@ -2387,7 +2387,16 @@ public IRubyObject name() {
 
     @JRubyMethod(name = "name")
     public IRubyObject name(ThreadContext context) {
-        return getBaseName() == null ? context.nil : rubyName().strDup(context.runtime);
+        if (getBaseName() == null) return context.nil;
+
+        Ruby runtime = context.runtime;
+        RubyString name = rubyName().strDup(runtime);
```

Comment:
name is already cached in a field. thus dup-ing and than freezing, not sure what the added benefit is.
this one either needs more work or should be dropped.
for now, I would vote for the later (and wait for Ruby to decide) ... we would real-world usage numbers.
like in case of the original PR @rdubya actually looked into the `Symbol#to_s` case being used a lot (in Rails).